src/Task/CloverCoverage.php

```diff
@@ -81,41 +112,96 @@ public function canRunInContext(ContextInterface $context): bool
+                'You must provide a positive minimum level between 1-100 for code coverage.'
             );
         }
 
-        $xml = new SimpleXMLElement($this->filesystem->readFromFileInfo(new SplFileInfo($cloverFile)));
-        $totalElements = (int) current($xml->xpath('/coverage/project/metrics/@elements') ?? []);
-        $checkedElements = (int) current($xml->xpath('/coverage/project/metrics/@coveredelements') ?? []);
+        try {
+            [
+                'totalElements' => $totalElements,
+                'checkedElements' => $checkedElements
+            ] = $this->parseTotals($cloverFiles, $mergeStrategy);
+        } catch (FileNotFoundException $exception) {
+            return TaskResult::createFailed($this, $context, $exception->getMessage());
+        }
 
         if (0 === $totalElements) {
             return TaskResult::createSkipped($this, $context);
         }
 
         $coverage = round(($checkedElements / $totalElements) * 100, 2);
 
-        if ($coverage < $percentage) {
+        if ($coverage < $minimumLevel) {
             $message = sprintf(
                 'Code coverage is %1$d%%, which is below the accepted %2$d%%'.PHP_EOL,
                 $coverage,
-                $percentage
+                $minimumLevel
             );
 
             return TaskResult::createFailed($this, $context, $message);
         }
 
+        if ($targetLevel !== null && $coverage < $targetLevel) {
+            $message = sprintf(
+                'Code coverage is %1$d%%, which is below the target %2$d%%'.PHP_EOL,
+                $coverage,
+                $targetLevel
+            );
+
+            return TaskResult::createNonBlockingFailed($this, $context, $message);
+        }
+
         return TaskResult::createPassed($this, $context);
     }
+
+    /**
+     * @param list<string> $coverageFiles
+     * @param CloverCoverage::MERGE_STRATEGY_* $mergeStrategy
+     * @return array{'totalElements': int, 'checkedElements': int}
+     *
+     * @throws FileNotFoundException
+     */
+    private function parseTotals(array $coverageFiles, string $mergeStrategy): array
+    {
+        $result = [
+            'totalElements' => 0,
+            'checkedElements' => 0,
+        ];
+
+        foreach ($coverageFiles as $file) {
+            if (!$this->filesystem->exists($file)) {
+                throw new FileNotFoundException($file);
+            }
+
+            $xml = new SimpleXMLElement($this->filesystem->readFromFileInfo(new SplFileInfo($file)));
+            $totalElements = (int) current($xml->xpath('/coverage/project/metrics/@elements') ?? []);
+            $checkedElements = (int) current($xml->xpath('/coverage/project/metrics/@coveredelements') ?? []);
+
+            $result = [
+                'totalElements' => match ($mergeStrategy) {
+                    self::MERGE_STRATEGY_COMBINE => $result['totalElements'] + $totalElements,
```

Comment:
Hindside, this doesn't seem like a good idea: The covered elements could be the same in multiple files, making this merge not true.

Another way could be to use PHP reports and merge them together into a full clover coverage file with https://github.com/sebastianbergmann/phpcov

```
phpcov merge --clover full-coverage-report.xml coveragefolder
```

That however, would be out of scope for grumphp.
Maybe a better idea to create a grumphp extension package especially for this purpose?

Reply:
Moved to https://github.com/phpro/grumphp-combined-coverage-extension/